### ISSUE TITLE
"quotient by zero" error in Risch code

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#190]
Imported from SourceForge on 2024-07-02 08:18:36
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-01-15 19:34:48
Original: https://sourceforge.net/p/maxima/bugs/3274

---

Reported to mailing list 2017-01-13: "Failure in integrate()"
Thanks to Konstantin Stopani for this report.

Maxima version: "5.38.0"
Maxima build date: "2016-04-03 08:56:53"
Host type: "x86_64-apple-darwin15.4.0"
Lisp implementation type: "SBCL"
Lisp implementation version: "1.2.10",

the following code produces an error (Maxima output shown with
"[Maxima:]"):

~~~~
        expr: (%e^-(t/tau3)*(%e^(t/tau3+t/tau1)-%e^t*tau3))
          /((tau1*%e^(t/tau1)+%e^(t/tau1))
             *%e^(t/tau2)*tau3+%e^(t/tau2+t/tau1))$

        integrate(expr, t);

[Maxima:] `quotient' by `zero'
[Maxima:]  -- an error. To debug this try: debugmode(true);
~~~~

However, when names of all the constants are changed, all works:

~~~~
        integrate(subst([tau1=c1, tau2=c2, tau3=c3], expr), t);

[Maxima:] (%o1) (... skipped presumably correct expression ...)
~~~~

Also, the online maxima version on maxima-online.org (Maxima 5.21.1
GCL linux) seems to not produce the error.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-02 08:18:38
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-01-15 20:19:47
Original: https://sourceforge.net/p/maxima/bugs/3274/#2898

---

- **labels**: integrate, polynomials --> integrate, polynomials, risch
- **summary**: "quotient by zero" error in integrate --> "quotient by zero" error in Risch code

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-02 08:18:40
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2017-01-15 20:19:47
Original: https://sourceforge.net/p/maxima/bugs/3274/#3cdc

---

The error is in RISCHLOGEPROG, which calls RATQU with a divisor equal to `(0 . 1)` (a representation of zero).

I have fixed this error by having RISCHLOGEPROG give up and return (without an error), so that `integrate` (via TRYRISCH) can try another rule, which succeeds.

Fixed by commit a7239b10. Closing this report.
